### PR TITLE
Ensure dev environment uses Arrow GPU packages

### PR DESCRIPTION
**Pull request by charlesbluca**
With #7495 merged, it seems like the dev environment files create an environment with the CPU packages for `pyarrow` and `arrow-cpp`; this results in failure when trying to compile libcudf or `import cudf`:

```python
ModuleNotFoundError: No module named 'pyarrow._cuda'
    from cudf import rmm
  File "/opt/conda/lib/python3.8/site-packages/cudf/__init__.py", line 11, in <module>
    from cudf import core, datasets, testing
  File "/opt/conda/lib/python3.8/site-packages/cudf/core/__init__.py", line 3, in <module>
    from cudf.core import _internals, buffer, column, column_accessor, common
  File "/opt/conda/lib/python3.8/site-packages/cudf/core/_internals/__init__.py", line 3, in <module>
    from cudf.core._internals.where import where
  File "/opt/conda/lib/python3.8/site-packages/cudf/core/_internals/where.py", line 11, in <module>
    from cudf.core.column import ColumnBase
  File "/opt/conda/lib/python3.8/site-packages/cudf/core/column/__init__.py", line 3, in <module>
    from cudf.core.column.categorical import CategoricalColumn
  File "/opt/conda/lib/python3.8/site-packages/cudf/core/column/categorical.py", line 25, in <module>
    from cudf import _lib as libcudf
  File "/opt/conda/lib/python3.8/site-packages/cudf/_lib/__init__.py", line 4, in <module>
    from . import (
ImportError: libarrow_cuda.so.400: cannot open shared object file: No such file or directory
```

This updates the dev environments and recipe to ensure that the GPU package of `pyarrow` (and `arrow-cpp` accordingly) are used.

### CODE REVIEW

**Review by raydouglass — CHANGES_REQUESTED** [review]
Can you also update https://github.com/rapidsai/cudf/blob/branch-21.08/conda/recipes/libcudf/meta.yaml#L40

**Comment by charlesbluca** [comment]
Do you know if the formatting for the recipe is correct? Not sure if the syntax differs significantly from the conda env file

File: conda/recipes/cudf/meta.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ requirements:
     - setuptools
     - numba >=0.53.1
     - dlpack>=0.5,<0.6.0a0
-    - pyarrow 4.0.1
+    - pyarrow 4.0.1=*cuda
```
Comment (jakirkham) [inline conda/recipes/cudf/meta.yaml:33]:
```suggestion
    - pyarrow 4.0.1 *cuda
```

**Review by ajschmidt8 — APPROVED** [review]

**Comment by jakirkham** [comment]
@gpucibot merge

**Comment by jakirkham** [comment]
Thanks all! 😄

**Comment by jakirkham** [comment]
Looks like we needed another change to the recipe. Added here ( https://github.com/rapidsai/cudf/pull/8651 )

**Pull request merged**